### PR TITLE
[IMP] project,hr_timesheet,sale_project,sale_timesheet,purchase: generic improvements for the project

### PR DESCRIPTION
**Pull request by matu-odoo**
Purpose of the PR is to do the generic improvements for project.

So in this PR did the following changes:
- adding a label 'last update' to the stat button in project form view.
- indicate 'my project' in the tab instead of 'project sharing view in portal' in project sharing.
- set the first non-folded stage of the project as default on newly created tasks.
- remove user confirming the SO as the default project manager.
- hide fields service_tracking, service_upsell_threshold if sale_ok is false.
- set purchase_method to purchase by default if product is of service type.
- remove the : next to the totals labels and decrease the font-size for values of 'total hours' 
  and 'remaining hours' in timesheets notebook in project task form view.

task-2897867

### CODE REVIEW

**Review by MissingNoShiny — COMMENTED** [review]
LGTM, thanks for your work :slightly_smiling_face:

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thank you! :fire: In case you didn't see it yet, specs have changed. Sorry about that.

File: addons/hr_timesheet/views/project_views.xml [hunk 1]
```diff
@@ -195,19 +195,22 @@
                                 <label class="fw-bold" for="effective_hours" string="Days Spent" attrs="{'invisible': [('encode_uom_in_days', '=', False)]}"/>
                             </span>
                             <field name="effective_hours" widget="timesheet_uom" nolabel="1"/>
+
```
Comment (auon-odoo) [inline addons/hr_timesheet/views/project_views.xml:198]: unnecessary change :angel:

File: addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js [hunk 2]
```diff
@@ -20,6 +20,16 @@ export class ProjectStatusWithColorSelectionField extends SelectionField {
         return this.colors[value] ? this.colorPrefix + this.colors[value] : "";
     }
 }
+
+SelectionField.props = {
+    ...SelectionField.props,
+    status_label: { type: String, optional: true },
+};
+
+ProjectStatusWithColorSelectionField.extractProps = ({ attrs }) =>{
```
Comment (auon-odoo) [inline addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js:29]:
```suggestion
ProjectStatusWithColorSelectionField.extractProps = ({ attrs }) => {
```
:angel:

File: addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js [hunk 3]
```diff
@@ -20,6 +20,16 @@ export class ProjectStatusWithColorSelectionField extends SelectionField {
...
+    return {...SelectionField.extractProps({attrs}), status_label: attrs.status_label};
```
Comment (auon-odoo) [inline addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js:30]:
```suggestion
    return {...SelectionField.extractProps({attrs}), statusLabel: attrs.status_label};
```
And this is clearly a detail, but don't you think props should be named in `camelCase` (whereas attrs are ok in `snake_case`)? :thinking:

**Review by auon-odoo — APPROVED** [review]
Thank you for your work! Please give me your opinion or make the required changes before RD review! :fire:

File: addons/purchase/models/product.py [hunk 4]
```diff
@@ -15,11 +15,17 @@ class ProductTemplate(models.Model):
     purchase_method = fields.Selection([
         ('purchase', 'On ordered quantities'),
         ('receive', 'On received quantities'),
-    ], string="Control Policy", help="On ordered quantities: Control bills based on ordered quantities.\n"
-        "On received quantities: Control bills based on received quantities.", default="receive")
+    ], string="Control Policy", compute='_compute_purchase_method', store=True, readonly=False,
+        help="On ordered quantities: Control bills based on ordered quantities.\n"
+            "On received quantities: Control bills based on received quantities.")
```
[the remark this answers is not in the document]
Comment (adjo-odoo) [reply addons/purchase/models/product.py:20]: Yes this is the part of spec,
https://nimb.ws/hdEPne product form view: if the product is of service type, set purchase_method to purchase by default.

File: addons/product/views/product_views.xml [hunk 5]
```diff
@@ -64,7 +64,7 @@
                                 <group name="group_general">
                                     <field name="active" invisible="1"/>
                                     <field name="detailed_type"/>
-                                    <field name="product_tooltip" string="" class="fst-italic text-muted"/>
+                                    <field name="product_tooltip" string="" class="fst-italic text-muted" attrs="{'invisible': ['&amp;', ('type', '=', 'service'), ('sale_ok', '=', False)]}"/>
```
Comment (xavierbol) [inline addons/product/views/product_views.xml:67]: Equivalent to: 

```suggestion
                                    <field name="product_tooltip" string="" class="fst-italic text-muted" attrs="{'invisible': [('type', '=', 'service'), ('sale_ok', '=', False)]}"/>
```
? 🤔

File: addons/hr_timesheet/views/project_project_views.xml [hunk 6]
```diff
@@ -65,11 +65,10 @@
                 </xpath>
                 <xpath expr="//div[hasclass('o_project_kanban_boxes')]" position="after">
                     <t t-set="badgeColor" t-value="'border-success'"/>
-                    <t t-set="badgeColor" t-value="'border-warning'" t-if="record.remaining_hours.raw_value / record.allocated_hours.raw_value &lt; 0.2 and record.remaining_hours.raw_value / record.allocated_hours.raw_value &gt; 0"/>
-                    <t t-set="badgeColor" t-value="'border-danger'" t-elif="record.remaining_hours.raw_value &lt; 0"/>
+                    <t t-set="badgeColor" t-value="'border-danger'" t-if="record.remaining_hours.raw_value &lt; 0"/>
                     <t t-set="title" t-value="'Remaining days'" t-if="record.encode_uom_in_days.raw_value"/>
                     <t t-set="title" t-value="'Remaining hours'" t-else=""/>
-                    <div t-if="record.allow_timesheets.raw_value and record.allocated_hours.raw_value &gt; 0"
+                    <div t-if="record.allocated_hours.raw_value &gt; 0"
```
Comment (xavierbol) [inline addons/hr_timesheet/views/project_project_views.xml:71]: Can you explain why did you do that? 🧐 🙂 

I did not find the point for that. 😇

File: addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js [hunk 7]
```diff
@@ -20,6 +20,16 @@ export class ProjectStatusWithColorSelectionField extends SelectionField {
...
 }
+
+SelectionField.props = {
+    ...SelectionField.props,
+    statusLabel: { type: String, optional: true },
+};
+
+ProjectStatusWithColorSelectionField.extractProps = ({ attrs }) => {
+    return {...SelectionField.extractProps({attrs}), statusLabel: attrs.status_label};
+};
```
Comment (xavierbol) [inline addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js:31]: Should be defined in `projectStatusWithColorSelectionField` object.

File: addons/sale_project/views/product_views.xml [hunk 8]
```diff
@@ -12,7 +12,7 @@
             </field>
             <field name="invoice_policy" position="after">
                 <field name="service_policy" string="Invoicing Policy" attrs="{'invisible': [('type', '!=', 'service')], 'required': [('type', '=', 'service')]}"/>
-                <field name="service_tracking" required="1" attrs="{'invisible': [('type', '!=', 'service')]}"/>
+                <field name="service_tracking" required="1" attrs="{'invisible': ['&amp;', ('type', '=', 'service'), ('sale_ok', '=', False)]}"/>
```
Comment (xavierbol) [inline addons/sale_project/views/product_views.xml:15]: Equivalent to:
```suggestion
                <field name="service_tracking" required="1" attrs="{'invisible': [('type', '=', 'service'), ('sale_ok', '=', False)]}"/>
```

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left some comments. 🙂

File: addons/hr_timesheet/views/project_project_views.xml (hunk 6 above)
Comment (ppr-odoo) [reply addons/hr_timesheet/views/project_project_views.xml]: > Can you explain why did you do that? monocle_face slightly_smiling_face
> 
> I did not find the point for that. innocent


I have reverted this because there is no mentioned  in task about this. I discussed with @adjo-odoo.he told me that this changes made by mistake.thanks

File: addons/sale_project/views/product_views.xml (hunk 8 above)
Comment (ppr-odoo) [reply addons/sale_project/views/product_views.xml]: > Equivalent to:

Done

File: addons/product/views/product_views.xml (hunk 5 above)
Comment (ppr-odoo) [reply addons/product/views/product_views.xml]: > Equivalent to:
> 
> ? thinking



> Equivalent to:
> 
> ? thinking



> Equivalent to:
> 
> ? thinking

Done

File: addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js [hunk 9]
```diff
@@ -20,11 +20,22 @@ export class ProjectStatusWithColorSelectionField extends SelectionField {
         return this.colors[value] ? this.colorPrefix + this.colors[value] : "";
     }
 }
+
+SelectionField.props = {
```
Comment (xavierbol) [inline addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js:24]:
```suggestion
ProjectStatusWithColorSelectionField.props = {
```

File: addons/project/views/project_task_views.xml [hunk 10]
```diff
@@ -287,7 +287,7 @@
                                 <span class="o_stat_text">in Recurrence</span>
                             </div>
                         </button>
-                        <button name="%(project_task_action_sub_task)d" type="action" class="oe_stat_button" icon="fa-tasks"
+                        <button name="project_task_action_sub_task" type="object" class="oe_stat_button" icon="fa-tasks"
```
Comment (xavierbol) [inline addons/project/views/project_task_views.xml:290]: Why not just adding the `default_project_id` in the context here instead of creating a new action for that? 🤔

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left one comment. 🙂

File: addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js [hunk 11]
```diff
@@ -20,11 +20,22 @@ export class ProjectStatusWithColorSelectionField extends SelectionField {
...
 ProjectStatusWithColorSelectionField.template = 'project.ProjectStatusWithColorSelectionField';
 
 export const projectStatusWithColorSelectionField = {
     ...selectionField,
     component: ProjectStatusWithColorSelectionField,
+    extractProps: ({ attrs }) => {
+        return {
+            statusLabel: attrs.status_label,
+        };
+    },
```
Comment (xavierbol) [inline addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js:38]: I did not see but there are also an `extractProps` function in `selectionField`. And so we should do this:
```suggestion
    extractProps: (fieldInfo) => {
        const props = selectionField.extractProps(fieldInfo);
        props.statusLabel = fieldInfo.attrs.status_label;
        return props;
    },
```

Comment (ppr-odoo) [reply addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js]: Your suggestion didn't work so I made some changes to your suggestion.okay for you ?

       extractProps({ options }) {
          const props = selectionField.extractProps(...arguments);
          props.statusLabel = options.status_label;
          return props;
      }

Comment (xavierbol) [reply addons/project/static/src/components/project_status_with_color_selection/project_status_with_color_selection_field.js]: Oh it is because I forgot to pass dynamicLabel in parameter for the parent function. 

If it does not also work then yes we can do as you suggest.

**Review by xavierbol — APPROVED** [review]
LGTM, thanks again for your work. 🙂 

robodoo r+
